napari/utils/_magicgui.py

```diff
@@ -16,21 +16,132 @@
 
 import weakref
 from functools import cache, partial
-from typing import TYPE_CHECKING, Any, get_args
+from typing import TYPE_CHECKING, Any, Optional, get_args
+
+import numpy as np
+from magicgui.widgets import ComboBox, FunctionGui
 
 from napari.utils._proxies import PublicOnlyProxy
+from napari.utils.notifications import show_info
 
 if TYPE_CHECKING:
     from concurrent.futures import Future
 
-    from magicgui.widgets import FunctionGui
-    from magicgui.widgets._bases import CategoricalWidget
+    from magicgui.widgets.bases import CategoricalWidget
 
     from napari._qt.qthreading import FunctionWorker
     from napari.layers import Layer
     from napari.viewer import Viewer
 
 
+def _get_layer_from_widget(gui: ComboBox, viewer: Viewer) -> Optional[Layer]:
+    """Retrieve layer used as input to function wrapped into magicgui .
+
+    Parameters
+    ----------
+    gui : magicgui.widgets.ComboBox
+        The instantiated ComboBox widget.
+
+    Returns
+    -------
+    Layer
+        layer passed as input to the function wrapped into magicgui.
+
+    """
+    from napari.layers import Layer
+
+    if isinstance(gui.value, Layer):
+        return gui.value
+
+    if gui.annotation is None or gui.annotation.__module__ != 'napari.types':
+        return None
+
+    layer_name = gui.current_choice[: -len(' (data)')]
+    return viewer.layers[layer_name]
+
+
+def _get_layers_from_widget(gui: FunctionGui, viewer: Viewer) -> list[Layer]:
+    """Retrieve layers used as input to function wrapped into magicgui .
+
+    Parameters
+    ----------
+    gui : magicgui.widgets.FunctionGui
+        The instantiated FunctionGui widget.
+
+    Returns
+    -------
+    list
+        list of layers passed as input to the function wrapped into magicgui.
+
+    """
+    res = []
+    for widg in gui:
+        if isinstance(widg, ComboBox):
+            layer = _get_layer_from_widget(widg, viewer)
+            if layer is not None:
+                res.append(layer)
+    return res
+
+
+def _get_spatial_from_layer(layer, ndim):
+    axis = np.array(range(-ndim, 0))
+
+    data2physical = layer._transforms['data2physical'].set_slice(axis)
+    return {
+        'scale': data2physical.scale,
+        'translate': data2physical.translate,
+        'rotate': data2physical.rotate,
+        'shear': data2physical.shear,
+        'affine': layer.affine.set_slice(axis).affine_matrix,
+    }
+
+
+def _cmp_meta(meta1, meta2):
```

Comment:
```suggestion
def _compute_meta(meta1, meta2):
```

Does the abbreviation "cmp" = "compute"?

Reply:
`compare` but maybe I should switch to `eq`.

Reply:
I think `_compare_meta` is fine.